mapboxDriver.js

```diff
@@ -1,79 +1,73 @@
 const delay = ms => new Promise(res => setTimeout(res, ms));
 
-const RX = /^(.*)access_token=(.*)$/;
-
-// GEOGRAPHIC BOUNDS: latitude/longitude, in degrees
-// TODO: Figure out what these should be from Mapbox API
-// getBoundsFromApi below was an attempt, but the bounds returned
-// from that endpoint covered the entire world map
-
-// For https://projects.propublica.org/miseducation/ page
-const northernBound = 49.382808;
-const southernBound = 24.521208;
-const easternBound = -66.945392;
-const westernBound = -124.736342;
-
-// For testing: Massachusetts
-// const northernBound = 42.886589;
-// const southernBound = 41.237964;
-// const easternBound = -69.928393;
-// const westernBound = -73.508142;
-
-// TODO: Figure out what these should be from Mapbox API
-const minZoomLevel = 2;
-const maxZoomLevel = 10;
-
-// Time to wait after each API call to Mapbox, in milliseconds (ms)
 const waitIntervalMs = 50;
 
 // Increment for each cardinal direction change, in degrees
 const fetchTileIncrement = 1;
 
-// TODO: Figure out what these should be from Mapbox API
-const tileSets = [
-  "mapbox.mapbox-streets-v7,propublica.schools-countries,mapbox.mapbox-terrain-v2,propublica.schools-states",
-  "propublica.opp_gap-districts-black",
-  "propublica.opp_gap-districts-hispanic"
-];
-
-const tilesBaseUrlA = "https://a.tiles.mapbox.com/v4/";
-const tilesBaseUrlB = "https://b.tiles.mapbox.com/v4/"
-const tileFormat = ".vector.pbf";
-
-
 export default async ({data, page, crawler}) => {
   await page.setRequestInterception(true);
 
-  let accessToken = "";
-  
+  let mapBoxURLs = [];
+  let fontsURLs = [];
+
   page.on("request", request => {
     const url = request.url();
-
-    if (accessToken.length === 0) {
-      const tokenAttempt = getAccessToken(url);
-      if (tokenAttempt.length > 0) {
-        accessToken = tokenAttempt;
-      }
+    if (url.includes(".json")) {
+      mapBoxURLs.push(url);
+    }
+    if (url.includes("fonts")) {
+      fontsURLs.push(url);
     }
-    
     request.continue();
   });
 
   await crawler.loadPage(page, data);
 
-  await fetchTiles(page, accessToken);
+  const zoomButton = await page.$("button.mapboxgl-ctrl-icon:nth-child(1)");
+  // Zoom in 3 times to get all the font names
+  await zoomButton.click();
+  await zoomButton.click();
+  await zoomButton.click();
+
+  await fetchFonts(page, fontsURLs);
+  await fetchTiles(page, mapBoxURLs);
 };
 
-function getAccessToken(url) {
-  // Get access token from Mapbox URL
-  const m = url.match(RX);
-  if (!m) {
-    return "";
+async function getMapBoxJson(url) {
+  console.log(`getting json for ${url}`);
+  return await fetch(url).then((response) => {
+    return response.json();
+  });
+}
+
+function dedupeFonts(fontsURLs) {
+  const setOfFonts = new Set();
+  for (let fontIndex = 0; fontIndex < fontsURLs.length; fontIndex++) {
+    let font = fontsURLs[fontIndex];
+    font = font.replace(/[0-9]{1,5}-[0-9]{1,5}\.pbf/, "{i}-{j}.pbf");
```

Comment:
regex to match font segments which documented here: https://docs.mapbox.com/api/maps/fonts/